docs/DeveloperGuide.md

```diff
@@ -1,13 +1,91 @@
+5. Navigate to `File > Project Structure > Project Settings > Project` and set `SDK` to `11`.
+6. Verify the setup process by running `MoneyGoWhere#main()`.
+You should see the following greeting message if the project setup is successful:
+
+```
+  __  __                         _____   __          ___                   
+ |  \/  |                       / ____|  \ \        / / |                  
+ | \  / | ___  _ __   ___ _   _| |  __  __\ \  /\  / /| |__   ___ _ __ ___ 
+ | |\/| |/ _ \| '_ \ / _ \ | | | | |_ |/ _ \ \/  \/ / | '_ \ / _ \ '__/ _ \
+ | |  | | (_) | | | |  __/ |_| | |__| | (_) \  /\  /  | | | |  __/ | |  __/
+ |_|  |_|\___/|_| |_|\___|\__, |\_____|\___/ \/  \/   |_| |_|\___|_|  \___|
+                           __/ |                                           
+                          |___/                                            
+
+Your MoneyGoWhere? Let me help you track it.
+```
+
+## Design & Implementation
+### Software Architecture:
+The software architecture diagram below describes the application's design and the interaction between components.
+
+![Software-Architecture](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/SoftwareArchitecture.puml)
+
+### Core Components:
+
+| Component     | Function                                                                            |
+|---------------|-------------------------------------------------------------------------------------|
+| MoneyGoWhere  | Main entrypoint of the application.                                                 |
+| Common        | Defines various parameters used by the application.                                 |
+| Exceptions    | Defines exceptions thrown by the program.                                           |
+| UserInterface | Provides functions to interface with the user via standard I/O and handle commands. |
+| Command       | Defines the commands accepted by the program along with its arguments.              |
+| Parser        | Provides functions to parse inputs read from standard input.                        |
+| Data          | Stores data and provides functions to operate on data.                              |
+| Storage       | Defines functions to save and load data.                                            |
+| Logger        | Defines functions to log the user's actions and the application's behaviour.        |
+
+### Component Interactions:
+The sequence diagram below describes the interaction between the various core components when a command is entered.
+In this example, 
+the user enters the command `Add-Expense -n Expense -a 7.80` to add an expense with the name `Expense` and the amount `7.80`.
+
+![Component-Interaction-On-Command-Entered](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/ComponentInteractionsOnCommandEntered.puml)
```

Comment:
Should the following method call be directed at the start of a new activation bar?

![image](https://user-images.githubusercontent.com/89023959/197943779-d99ce72e-6b06-47dc-b412-4298981db476.png)